### DISCUSSION TITLE
Help with Dual-NIC config for Unraid + Frigate + External Home Assistant

### DISCUSSION THREAD

**Discussion by krcm0209** [blakeblackshear/frigate#3838] [Ask A Question]
I'm no networking expert, but I think what I want to do is possible, and am seeking help with the software/networking configuration to get it all working.

Here's a diagram of my physical network:
![image](https://user-images.githubusercontent.com/17315774/190226867-2e7b3d2e-240c-4f9b-86e2-435eeddb3b64.png)

1. How would I configure unRaid to allow for the Frigate docker container running on the unRaid machine to see the IP cameras?
2. How do can the IP cameras be assigned IP addresses? Do I need to add a docker container for Pi-Hole or a similar application that can act as a DHCP server / software router? If so, what are the options for that?
3. Are there special considerations for ensuring the Frigate container can communicate with the MQTT container, and by extension, the Raspberry Pi which is connected to a different network card than the cameras/PoE switch?
4. Are VLANs important here or no? I would assume a VLAN might be useful if I only had one physical NIC and wanted better separation of traffic. Is that right?

Thanks

**Comment by digitalit** [comment]
@krcm0209 as i use to start with, im absolute no expert but i have tryed a few ways of running Frigate and this looks a bit complicated. Maby there are some high tech reason my lack of skills prevents me from seing.

Anyway, i use an unRaid server and there i have all my storing so i don´t understand the Mediasonic part but please tell if i´ve missed a cooler way regarding storing.

Regarding Frigate i were looking for a good NVR for my Home Assisstant. Anyway i tried running Frigate in docker on my unRaid server but i really never got it working as i wanted so i run Home Assistant in a VM on my unRaid server and the VM has  [Hassos](https://github.com/home-assistant/operating-system) installed and the addon HACS that lets me install Frigate in Home Assistant and it works great. I also have an recommended Google Coral connected with USB to my unRaid server that with passtrough to my VM makes Frigate works even better.

Have a nice day and please forgive me if this was a useless answer to your problem

**Comment by blakeblackshear** [comment]
I haven't used unraid, but I did this on Debian. All I did was assign a static IP to NIC 2 on a different subnet than the main network. Cameras were assigned static IP addresses in the same subnet. No need to route traffic between subnets because I don't want the cameras to be able to reach the internet anyway.

**Comment by NickM-27** [comment]
I run Unraid with Frigate in docker. In Unraid I've typically seen multiple NICs used as redundant connections in case one goes bad so I know how to set it up like that, but Unraid has a lot of options for network setups so I think it should definitely be doable the way you want. 

I think you would just need to add another route table in the network setup, I may be able to help more at the time you're trying to set it up. I'm sure there's some Unraid experts that could make it happen pretty quick if you posted on the forum.
